api/app/Repositories/ElasticSearch/ElasticSearchRepository.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories\ElasticSearch;
+
+use App\Entity\Event;
+use App\Repositories\BaseRepository;
+
+final class ElasticSearchRepository extends BaseRepository implements ElasticSearchRepositoryInterface
+{
+    public function createIndex(array $data): void
```

Comment:
Name createIndex misleading.  Inside  we call ES  function index that means make index of items and save in your ElasticRepository. In lines 17, 22,27 name of your function is same as called from facade Elasticsearch maybe in that place you may use same politics with naming,